[deps]: update jwt-go dependency

emirisman opened the pull request:
## Description

jwt-go has been renamed to jwt and has a new home. See
https://github.com/dgrijalva/jwt-go/issues/462

## Checklist:
- [x] Documentation updated

harshavardhana commented on CREDITS:
```diff
@@ -3770,10 +3770,11 @@ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+https://github.com/golang-jwt/jwt
 ----------------------------------------------------------------
 Copyright (c) 2012 Dave Grijalva
+Copyright (c) 2021 jwt-go maintainers
```
is this manually edited? @emirisman

emirisman replied:
It is the current LICENSE file https://github.com/golang-jwt/jwt/blob/main/LICENSE

emirisman replied:
oh my bad, copied the wrong string

harshavardhana commented on cmd/jwt.go:
```diff
@@ -22,8 +22,8 @@ import (
 
-	jwtgo "github.com/dgrijalva/jwt-go"
-	jwtreq "github.com/dgrijalva/jwt-go/request"
+	jwtgo "github.com/golang-jwt/jwt"
+	jwtreq "github.com/golang-jwt/jwt/request"
```
Before we do this, I think the `golang-jwt/jwt` should fix the incompatible issue with go.mod

emirisman replied:
They have added go modules support but had to revert it. See https://github.com/golang-jwt/jwt/pull/26 and https://github.com/golang-jwt/jwt/issues/17 for details. It doesn't have any external dependencies itself therefore there are no downsides except the visually unpleasing `+incompatible` suffix

harshavardhana commented on go.sum:
```diff
@@ -533,6 +533,8 @@ github.com/gogo/protobuf v1.2.2-0.20190723190241-65acae22fc9d/go.mod h1:SlYgWuQ5
 github.com/gogo/protobuf v1.3.1/go.mod h1:SlYgWuQ5SjCEi6WLHjHCa1yvBfUnHcTbrrZtXPKa29o=
 github.com/gogo/protobuf v1.3.2 h1:Ov1cvc58UF3b5XjBnZv7+opcTcQFZebYjWzi34vdm4Q=
 github.com/gogo/protobuf v1.3.2/go.mod h1:P1XiOD3dCwIKUDQYPy72D8LYyHL2YPYrpS2s69NZV8Q=
+github.com/golang-jwt/jwt v3.2.1+incompatible h1:73Z+4BJcrTC+KczS6WvTPvRGOp1WmfEP4Q1lOd9Z/+c=
+github.com/golang-jwt/jwt v3.2.1+incompatible/go.mod h1:8pz2t5EyA70fFQQSrl6XZXzqecmYZeUEB8OUGHkxJ+I=
 github.com/golang-sql/civil v0.0.0-20190719163853-cb61b32ac6fe/go.mod h1:8vg3r2VgvsThLBIFL93Qb5yWzgyZWhEmBwUJWevAkK0=
 github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b h1:VKtxabqXZkF25pY9ekfRL6a582T4P37/31XEstQ5p58=
 github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b/go.mod h1:SBH7ygxi8pfUlaOkMMuAQtPIUF8ecWP5IEl/CR7VP2Q=
```
can you run `go mod tidy` ?

emirisman replied:
It is included, `go mod tidy` doesn't make further changes

harshavardhana replied:
Okay..

emirisman committed:
[deps]: update jwt-go dependency

jwt-go has been renamed to jwt and has a new home. See
https://github.com/dgrijalva/jwt-go/issues/462

klauspost reviewed (approved):
LGTM
